core/src/main/java/hudson/util/RobustReflectionConverter.java

```diff
@@ -290,6 +290,53 @@ public Object unmarshal(final HierarchicalStreamReader reader, final Unmarshalli
         return serializationMethodInvoker.callReadResolve(result);
     }
 
+    private void processAttribute(final Object result, final HierarchicalStreamReader reader, final SeenFields seenFields, String attrAlias) {
+        String attrName = mapper.attributeForAlias(attrAlias);
+        Class classDefiningField = determineWhichClassDefinesField(reader);
+        boolean fieldExistsInClass = fieldDefinedInClass(result, attrName);
+        if (fieldExistsInClass) {
+            Field field = reflectionProvider.getField(result.getClass(), attrName);
+            SingleValueConverter converter = mapper.getConverterFromAttribute(field.getDeclaringClass(), attrName, field.getType());
+            Class type = field.getType();
+            if (converter == null) {
+                converter = mapper.getConverterFromItemType(type);
+                // TODO add fieldName & definedIn args
+            }
+            if (converter != null) {
+                Object value = converter.fromString(reader.getAttribute(attrAlias));
+                if (type.isPrimitive()) {
+                    type = Primitives.box(type);
+                }
+                if (value != null && !type.isAssignableFrom(value.getClass())) {
+                    throw new ConversionException("Cannot convert type " + value.getClass().getName() + " to type " + type.getName());
+                }            reflectionProvider.writeField(result, attrName, value, classDefiningField);
```

Comment:
Best to preserve the formatting in this type of refactoring.  I was surprised to see that the formatting was changed on this line.
```suggestion
                }
                reflectionProvider.writeField(result, attrName, value, classDefiningField);
```